### PR TITLE
Documentation: Add Comments to all Functions in header files

### PR DESCRIPTION
**Pull request by RisingLeaf**
### Description
This PR's intention is to make editing ES code easier and more accessable by adding a comment in front of (nearly) every function in header files.
While some functions or group of functions already have some smaller or larger comments describing them, a lot are standing there without. The idea is to have a unfied comment format in front of them, that explains what they do and where they are accessed.

### Format
The format I thought of is best presented by this example:
```
// Disabled or stranded ships coordinate with other ships to get assistance. //
// Function to check if the ship is being helped, and if not, ask for help.
// Called during an AI step if the ship is stranded or disabled.
void AskForHelp(Ship &ship, bool &isStranded, const Ship *flagship);
// Function to check if the given ship can be helped by the helper ship.
// Called when asking for help and when checking if the ship has a helper.
static bool CanHelp(const Ship &ship, const Ship &helper, const bool needsFuel);
// Function to check if the given ship already has a helper ship.
// Called when asking for help.
bool HasHelper(const Ship &ship, const bool needsFuel);
```
Here I added a small heading for this group of functions and then a detailed description.

### Help Wanted
Of course I will be working on this, but it is a tiring process that takes a lot of time. Here is where you come in:
Want to write some descriptions? Do it and PR them to my fork of ES. I will be grateful for every help I get.

### Progress
For now I only added comments to the AI class, to give an example.

### CODE REVIEW

File: source/AI.h [hunk 1]
```diff
@@ -51,50 +51,94 @@ class AI {
...
 	void Clean();
-	// Clear ship orders. This should be done when the player lands on a planet,
-	// but not when they jump from one system to another.
+	// Function to clear ship orders. This should be done when the player
+	// lands on a planet, but not when they jump from one system to another.
+	// Called when ships are "placed" in a system for example on enter.
 	void ClearOrders();
-	// Issue AI commands to all ships for one game step.
+	// Function to issue AI commands to all ships for one game step.
+	// Called when the ENgine calculates a Step.
```
Comment (tibetiroka) [inline source/AI.h:87]:
```suggestion
	// Called when the Engine calculates a Step.
```

File: source/AI.h [hunk 2]
```diff
@@ -120,35 +164,46 @@ template <class Type>
...
 	// Return a bitmask giving the weapons to fire.
+	// Called in an AI step and when moving the player.
 	void AutoFire(const Ship &ship, FireCommand &command, bool secondary = true) const;
 	void AutoFire(const Ship &ship, FireCommand &command, const Body &target) const;
 
-	// Calculate how long it will take a projectile to reach a target given the
+	// Function to calculate how long it will take a projectile to reach a target given the
 	// target's relative position and velocity and the velocity of the
 	// projectile. If it cannot hit the target, this returns NaN.
+	// Called in all kind of movement functions.
```
Comment (tibetiroka) [inline source/AI.h:190]:
```suggestion
	// Called in all kinds of movement functions.
```

File: source/AI.h [hunk 3]
```diff
@@ -120,35 +164,46 @@ template <class Type>
...
 	static double RendezvousTime(const Point &p, const Point &v, double vp);
 
+	// Move the players ship on basis of the given commands.
+	// Called in an Ai step.
```
Comment (tibetiroka) [inline source/AI.h:194]:
```suggestion
	// Called in an AI step.
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]
Quick grammar review. This review does not address the format itself.

File: source/AI.h [hunk 4]
```diff
@@ -51,50 +51,94 @@ class AI {
...
 	using List = std::list<std::shared_ptr<Type>>;
+
 	// Constructor, giving the AI access to various object lists.
+	// Called on Engine Constructor.
 	AI(const List<Ship> &ships, const List<Minable> &minables, const List<Flotsam> &flotsam);
 
-	// Fleet commands from the player.
+	// Fleet commands from the player. //
+	// Fuction that sets a ship as the target of the players fleet.
+	// If the trageted ship is an enemy the fleet will be given the
```
Comment (1010todd) [inline source/AI.h:61]:
```suggestion
	// If the targeted ship is an enemy the fleet will be given the
```

**Comment by petervdmeer** [comment]
Good initiative, but if we do this, then let's use a standardized parseble format, for example something like [Doxygen](https://doxygen.nl/). such a standard format allows us to automatically generate API documentation, and people outside of ES might also have seen it before (easier/quicker to read), and such a standard format might also integrate better into IDEs.

Comments can be added in small batches; so I feel it is more efficient to let everybody just PR new sets of comments directly to the main branch (after you give a good example :) ). Collecting them in your own branch first increases the risk on merge-conflicts, while it doesn't add much in the process.

The [Design by Contract](https://github.com/endless-sky/endless-sky/issues/7015) idea also seems somewhat relevant, but I suggest we mostly ignore that in relation to comments until somebody actually sets up some design by contract support.

**Comment by RisingLeaf** [comment]
> Good initiative, but if we do this, then let's use a standardized parseble format, for example something like [Doxygen](https://doxygen.nl/). such a standard format allows us to automatically generate API documentation, and people outside of ES might also have seen it before (easier/quicker to read), and such a standard format might also integrate better into IDEs.

Hmmm... In this case I propose to use the following format:
```
/**
 * A brief history of JavaDoc-style (C-style) comments.
 *
 * This is the typical JavaDoc-style C-style comment. It starts with two
 * asterisks.
 *
 * @param theory Even if there is only one possible unified theory. it is just a
 *               set of rules and equations.
 */
void cstyle( int theory );
```

**Comment by RisingLeaf** [comment]
Closing this, because we probably need a structure of how we want to do this before starting to add comments everywhere.

**Pull request closed without merging**